### DISCUSSION TITLE
Question on optiontable.pl

### DISCUSSION THREAD

**Discussion by gvanem** [curl/curl#9990] [General]
I note that there are many differences in my generated `lib/easyoptions.c` compared
to what's in git master. These are some of the diffs right now:
```diff
--- my-version/lib/easyoptions.c     2022-11-28 14:46:17
+++ master-version/lib/easyoptions.c 2022-11-19 11:52:19
@@ -53,6 +53,12 @@
   {"CONNECTTIMEOUT_MS", CURLOPT_CONNECTTIMEOUT_MS, CURLOT_LONG, 0},
   {"CONNECT_ONLY", CURLOPT_CONNECT_ONLY, CURLOT_LONG, 0},
   {"CONNECT_TO", CURLOPT_CONNECT_TO, CURLOT_SLIST, 0},
+  {"CONV_FROM_NETWORK_FUNCTION", CURLOPT_CONV_FROM_NETWORK_FUNCTION,
+   CURLOT_FUNCTION, 0},
+  {"CONV_FROM_UTF8_FUNCTION", CURLOPT_CONV_FROM_UTF8_FUNCTION,
+   CURLOT_FUNCTION, 0},
+  {"CONV_TO_NETWORK_FUNCTION", CURLOPT_CONV_TO_NETWORK_FUNCTION,
+   CURLOT_FUNCTION, 0},
   {"COOKIE", CURLOPT_COOKIE, CURLOT_STRING, 0},
   {"COOKIEFILE", CURLOPT_COOKIEFILE, CURLOT_STRING, 0},
   {"COOKIEJAR", CURLOPT_COOKIEJAR, CURLOT_STRING, 0},
...
```

And that `lib/easyoptions.c` is *older* than `include/curl/curl.h` that it should be generated from.

I'd assume that some git-hook could run `perl lib/optiontable.pl include/curl/curl.h > lib/easyoptions.c` before
a commit.  Would that be an idea?

**Comment by bagder** [comment]
`optiontable.pl` needs to be adjusted after 6967571bf2062

**Reply by bagder** [reply]
Filed it as issue #9992

**Comment by gvanem** [comment]
I still have issues with generating `easyoptions.c` correctly with different Perls and line-endings.
E.g. with `core.autocrlf=true` in my global section of git, this Cygwin command:
`perl.exe ./optiontable.pl < ../include/curl/curl.h > easyoptions.c` gets broken.
But a Strawberry Perl works with both forms of line-endings.

So I tried to fix it. This works: 
`dos2unix ../include/curl/curl.h & perl.exe ./optiontable.pl < ../include/curl/curl.h > easyoptions.c`

But it seems better to patch `optiontable.pl`:
```diff
--- a/lib/optiontable.pl 2023-01-19 14:43:40
+++ b/lib/optiontable.pl 2023-06-10 11:59:06
@@ -55,6 +55,7 @@
     $ext =~ s/CURLOPTTYPE_//;
     $ext =~ s/CBPOINT/CBPTR/;
     $ext =~ s/POINT\z//;
+    $ext =~ s/\r//;
     $type = "CURLOT_$ext";

     $opt{$name} = $opt;
@@ -73,6 +74,7 @@
 my $fl;
 while(<STDIN>) {
     my $l = $_;
+    $l =~ s/[\r\n]//g;
     if($fl) {
         # continued deprecation
         if($l =~ /(.*)\),/) {
```

This seems to work with Cygwin+Strawberry Perl and  `core.autocrlf=false/true`.
I do not know much Perl. So someone else please comment.

**Reply by jay** [reply]
Doesn't cygwin come with its own perl, and if you are in cygwin then shouldn't it be expecting LF-only?

**Reply by gvanem** [reply]
> *and if you are in cygwin ...* 

Not sure I parse this. I never use Cygwin from an interactive shell like mintty/bash. Yuk!
I use my installed Cygwin programs from my normal shell (4NT). Works fine most times 
And my Cygwin `perl.exe`  is ahead of  my Strawberry `perl.exe`.

But fixing my [`/etc/fstab`](https://cygwin.com/cygwin-ug-net/using.html#mount-table) seems like a fix. Specify that all curl text-files should be `CRLF`  terminated:
`f:/MinGW32/src/inet/curl /cygdrive/f/MinGW32/src/inet/curl ntfs text,posix=0 0 0`
